src/server/api/AdminPanel.ts

```diff
@@ -0,0 +1,187 @@
+import type { Response } from "express";
+// @ts-ignore
+import { deleteUser, getMemberDataByCriteria, updateMemberColumns } from "../database/memberManager.js";
+// @ts-ignore
+import { deleteRefreshTokensOfUser } from "../database/refreshTokenManager.js";
+// @ts-ignore
+import { logEvents } from "../middleware/logEvents.js";
+// @ts-ignore
+import { deleteAccount } from "../controllers/deleteAccountController.js";
+// @ts-ignore
+import { deleteAllSessionsOfUser } from "../controllers/authenticationTokens/sessionManager.js";
+
+const validCommands = [
+	"ban",
+	"delete",
+	"username",
+	"logout",
+	"verify",
+	"post",
+	"invites",
+	"announce",
+	"userinfo",
+	"help"
+];
+
+function processCommand(req: CustomRequest, res: Response): void {
+	const command = req.params["command"]!;
+	const commandAndArgs = command.split(" ");
+	if (!req.memberInfo.signedIn) {
+		res.status(401).send("Cannot send commands while logged out.");
+		return;
+	}
+	if (!(req.memberInfo.roles?.includes("admin") ?? false)) {
+		res.status(403).send("Cannot send commands without the admin role");
+		return;
+	}
+	// TODO prevent affecting accounts with equal or higher roles
+	switch (commandAndArgs[0]) {
+		case "ban":
+			return;
+		case "delete":
+			deleteCommand(command, commandAndArgs, res);
+			return;
+		case "username":
+			usernameCommand(command, commandAndArgs, res);
+			return;
+		case "logout":
+			logoutUser(command, commandAndArgs, res);
+			return;
+		case "verify":
+			return;
+		case "post":
+			return;
+		case "invites":
+			return;
+		case "announce":
+			return;
+		case "userinfo":
+			getUserInfo(command, commandAndArgs, res);
+			return;
+		case "help":
+			helpCommand(commandAndArgs, res);
+			return;
+		default:
+			res.status(422).send("Unknown command.");
+			return;
+	}
+}
+
+function deleteCommand(command: string, commandAndArgs: string[], res: Response) {
+	logCommand(command);
+	if (commandAndArgs.length < 2) {
+		res.status(422).send("Invalid number of arguments, expected 1, got " + (commandAndArgs.length - 1) + ".");
+		return;
+	}
+	try {
+		deleteAccount(getMemberDataByCriteria(["user_id"], "username", commandAndArgs[1])["user_id"] ?? (() => { throw new Error(); })(), commandAndArgs[2] ?? "");
+	}
+	catch {
+		res.status(422).send("User " + commandAndArgs[1] + " does not exist.");
+		return;
+	}
+	res.status(200).send("");
+}
+
+function usernameCommand(command: string, commandAndArgs: string[], res: Response) {
+	if (commandAndArgs[1] === "get") {
+		if (commandAndArgs.length < 3) {
+			logCommand(command);
+			res.status(422).send("Invalid number of arguments, expected 2, got " + (commandAndArgs.length - 1) + ".");
+			return;
+		}
+		const parsedId = Number.parseInt(commandAndArgs[2]!);
+		logEvents("Command executed: " + command + "\nResult: " + getMemberDataByCriteria(["username"], "user_id", parsedId)["username"] + "\n", "adminCommands");
+		const username = getMemberDataByCriteria(["username"], "user_id", parsedId)["username"];
```

Comment:
Duplicate call of getMemberDataByCriteria() in the above 2 lines. Merge their result into one variable.